### PR TITLE
Add IPython-SQL to conda-forge

### PR DESCRIPTION
**Pull request by sodre**
Please add iPython-SQL to conda-forge.

### CODE REVIEW

File: recipes/ipython-sql/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{$ set version = "0.3.6" %}
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: Should be `{% set version = "0.3.6" %}`.

Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: Small punctuation is our downfall again. 😆

File: recipes/ipython-sql/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+test:
+  # Python imports
+  imports:
+    - sql
+
+about:
+  home: pypi.python.org/pypi/ipython-sql
+  license: MIT License
+  summary: 'RDBMS access via IPython'
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: Please add yourself as a maintainer. 😉

```yaml
extra:
  recipe-maintainers:
    - sodre
```

File: recipes/ipython-sql/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  run:
+    - python
+    - prettytable
+    - ipython >=1.0
+    - sqlalchemy >=0.6.7
+    - sqlparse
+    - six
+
+test:
+  # Python imports
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: nit: Please drop this comment.

File: recipes/ipython-sql/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  name: ipython-sql
+  version: {{ version }}
+
+source:
+  fn: ipython-sql-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/i/ipython-sql/ipython-sql-{{ version }}.tar.gz
+  md5: {{ md5 }}
+#  patches:
+   # List any patch files here
+   # - fix.patch
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: nit: Please drop these comments.

File: recipes/ipython-sql/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    - six
+
+test:
+  # Python imports
+  imports:
+    - sql
+
+about:
+  home: pypi.python.org/pypi/ipython-sql
+  license: MIT License
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: Simply `MIT` is fine here.

**Comment by jakirkham** [comment]
Looks pretty good @sodre. Few things to fix up, but they are all pretty easy. Thanks for the proposed addition.

**Comment by msarahan** [comment]
Circle failure appears related to trying to compile pexpect python3 file.  I think/hope newer versions of conda-build will fix this.  Not sure if you want to try with 1.21.11 from source, package seems to not be available yet.

**Comment by jakirkham** [comment]
Didn't `pip` also work in this case?

**Comment by sodre** [comment]
let me know what/if you need me to change.

File: recipes/ipython-sql/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  name: ipython-sql
+  version: {{ version }}
+
+source:
+  fn: ipython-sql-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/i/ipython-sql/ipython-sql-{{ version }}.tar.gz
+  md5: {{ md5 }}
+
+build:
+  number: 0
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: Well, if you want to just do Python 3.x, please add `skip: true  # [py2k]`. If we need Python 3.5 only, we may need to change `py2k` to `py35`. Once this is done, please drop the changes to `.CI/build_all` from the PR.

File: .CI/build_all [hunk 7]
```diff
@@ -23,9 +23,8 @@ if [ -z "$GH_TOKEN" ]; then
     # We don't need to build the example recipe.
     rm -rf ./recipes/example
 
-    # revert conda-build to 1.21.7, to get ObsPy 1.0.2 built, see conda-forge/staged-recipes#582
-    conda install --yes conda-build=1.21.7
+    conda install --yes conda-build=1.21.11
```
Comment (jakirkham) [inline .CI/build_all]: This has changed.

File: .CI/build_all [hunk 8]
```diff
@@ -23,9 +23,8 @@ if [ -z "$GH_TOKEN" ]; then
...
 
     # We just want to build all of the recipes.
-    conda-build-all ./recipes --matrix-condition "numpy >=1.10" "python >=2.7,<3|>=3.4"
+    conda-build-all ./recipes --matrix-condition "numpy >=1.10" "python <3|>=3.4"
```
Comment (jakirkham) [inline .CI/build_all]: This can be dropped thanks to the change noted below.

File: recipes/ipython-sql/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+skip:
+  true # [py2k]
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: So this should be in the `build` section, it should be one line, and the selector should have at least 2 spaces before it.

Comment (sodre) [inline recipes/ipython-sql/meta.yaml]: @jakirkham,
   Thanks so much for helping me out on this. is there an easy way to run the lint checker on my side before committing?

File: recipes/ipython-sql/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+source:
+  fn: ipython-sql-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/i/ipython-sql/ipython-sql-{{ version }}.tar.gz
+  md5: {{ md5 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  skip: true  # [py2k]
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: We should be able to drop this `skip` line as `conda-build` version `1.21.11` is out and shouldn't run into the same issues.

**Comment by jakirkham** [comment]
Hmm...seems we are still running into this issue on CircleCI still, but not the other CIs. Will try rebuilding the Docker image.

As for the other CIs, seems we may be running into some other sort of issue ( https://github.com/spyder-ide/spyder/issues/3309 ). @ccordoba12, do you know anything about what is causing the [`AttributeError: module 'pkg_resources' has no attribute 'iter_entry_points'`]( https://travis-ci.org/conda-forge/staged-recipes/jobs/150523796#L1300 ). It seems to only occur on Python 3.5. 😕

**Comment by jakirkham** [comment]
Interestingly I don't run into this issue with `setuptools` alone or with `pygments` for that matter. There is something more going on that seems to be missing.

**Comment by jakirkham** [comment]
Rebuilt the Docker container and now restarting CircleCI.

**Comment by jakirkham** [comment]
@conda-forge/pygments, any ideas on this last error?

**Comment by jakirkham** [comment]
As for the first error, could you please try updating `master` for the `conda-forge` remote and merging with it, @sodre? I expect this should fix it.

**Comment by ccordoba12** [comment]
@jakirkham, @Carreau said in spyder-ide/spyder#3309 that this problem was a bug in IPython 5.0, and that it'll be fixed in 5.1.

**Comment by jakirkham** [comment]
Ah, ok, sorry for being dense. Thanks for clarifying, @ccordoba12.

**Comment by Carreau** [comment]
Yes, Sorry I'm trying to push 5.1 / 5.0.1 through the door. I'm doing the best I can. Sorry about that.

**Comment by jakirkham** [comment]
No worries. Just trying to narrow down the cause.

**Comment by jakirkham** [comment]
If it would help alleviate some pressure, @Carreau, we could try applying a patch that fixes the issue to our `ipython` package. That is assuming the fix is known and available. Just let us know. 😄

**Comment by Carreau** [comment]
Ok, I had a deeper look and now I'm confused. Looking at the travis error, I have no clue why. 

The issue @ccordoba12 linked to have 2 issues. One with `AttributeError: module 'pkg_resources' has no attribute 'iter_entry_points'` which is usually resolved by upgrading setuptools. The other `ImportError: cannot import name 'NoOpContext'` which was fixed there: https://github.com/ipython/ipython/pull/9765

**Comment by jakirkham** [comment]
Was referring to the latter issue. Though we are using the latest `setuptools`. Maybe switching to `pip` fixes this.

**Comment by Carreau** [comment]
<joking>
Btw, I think it is failing because IPython has been spelled with a lower case i in a few places, while it is IPython
</joking>

**Comment by Carreau** [comment]
Dang, the `joking` tags are removed by GitHUb.

**Comment by jakirkham** [comment]
Must be GitHub's way of saying no more joking around. 😜

**Comment by sodre** [comment]
alright, it looks like the only issue left is with pygments for python 3.5, IPython 5 and win[64,32].
I will try with IPython 4 and see if it works for that platform.

**Comment by sodre** [comment]
@jakirkham, I looks like the package is now ready to go.
What is the next step from my side?

File: recipes/ipython-sql/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    - sqlalchemy >=0.6.7
+    - sqlparse
+    - six
+
+test:
+  imports:
+    - sql
+
+about:
+  home: pypi.python.org/pypi/ipython-sql
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: Might change to https://pypi.python.org/pypi/ipython-sql

File: recipes/ipython-sql/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    - sqlparse
+    - six
+
+test:
+  imports:
+    - sql
+
+about:
+  home: https://pypi.python.org/pypi/ipython-sql
+  license: MIT
```
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: Ideally we would include the license file. However, they don't provide one. 😞

Have raised issue ( https://github.com/catherinedevlin/ipython-sql/issues/58 ) to track this. Could we please add a comment here to add the [`license_file`]( http://conda.pydata.org/docs/building/meta-yaml.html#license-file ) once they have provided one?

File: recipes/ipython-sql/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/ipython-sql/meta.yaml]: Guessing this is not needed now, but have forgotten. Maybe a reminder would be nice please. 😁

**Comment by jakirkham** [comment]
Thanks everyone for your help getting this one worked out. Thanks @sodre for contributing it.

**Comment by jakirkham** [comment]
Soon you should get an email that invites you to join conda-forge, @sodre. Once accepted you will be added to a team with the same name as this recipe. Those will give you permissions on the feedstock (repo) for this recipe.

Make sure when proposing any change that you go through the typical GitHub workflow of forking the feedstock and making changes in your fork that you PR back. Once merged CIs will build and deploy any changes you make. Please let us know if you have any questions and welcome to conda-forge.

**Comment by sodre** [comment]
Thanks! I will keep an eye out for the License fix.

Best,
Patrick

**Pull request merged**